app/controllers/identify_controller.rb

```diff
@@ -63,14 +67,26 @@ def client_cert
 
   # :reek:UtilityFunction
   def process_cert(raw_cert)
-    cert = Certificate.new(OpenSSL::X509::Certificate.new(raw_cert))
+    x509_cert = OpenSSL::X509::Certificate.new(raw_cert)
+    cert = Certificate.new(x509_cert)
 
-    cert.token(nonce: nonce)
+    cert.token(nonce: nonce, has_eku: IdentifyController.certificate_has_eku?(x509_cert))
   rescue OpenSSL::X509::CertificateError => error
+    certificate_bad_error(error)
+  end
+
+  def certificate_bad_error(error)
     logger.warn("CertificateError: #{error.message}")
     TokenService.box(error: 'certificate.bad', nonce: nonce)
   end
 
+  def self.certificate_has_eku?(x509_cert)
```

Comment:
instead of keeping this logic on the controller, could we put this on our `models/certificate.rb` class? It already wraps/adds extra logic to x509 certs

Reply:
And then we could add specs for it too plz